plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/property/ConnectionPropertySource.java

```diff
@@ -8,18 +8,75 @@
  ******************************************************************************/
 package org.jboss.tools.openshift.internal.ui.property;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.part.IPage;
 import org.eclipse.ui.views.properties.IPropertyDescriptor;
 import org.eclipse.ui.views.properties.IPropertySource;
+import org.eclipse.ui.views.properties.PropertySheet;
+import org.eclipse.ui.views.properties.PropertySheetPage;
+import org.eclipse.ui.views.properties.tabbed.TabbedPropertySheetPage;
+import org.jboss.tools.openshift.common.core.connection.ConnectionsRegistrySingleton;
 import org.jboss.tools.openshift.common.core.connection.IConnection;
+import org.jboss.tools.openshift.common.core.connection.IConnectionsRegistryListener;
+import org.jboss.tools.openshift.core.ICommonAttributes;
+import org.jboss.tools.openshift.core.connection.IOpenShiftConnection;
 
 public class ConnectionPropertySource implements IPropertySource {
 	
 	private static final String HOST = "host";
 	private static final String USERNAME = "username";
 	private IConnection connection;
 
+	private ConnectionListener listener = new ConnectionListener();
+
+	class ConnectionListener implements IConnectionsRegistryListener {
+
+		@Override
+		public void connectionAdded(IConnection connection) {
+		}
+
+		@Override
+		public void connectionRemoved(IConnection connection) {
+		}
+
+		@Override
+		public void connectionChanged(IConnection connection, String property, Object oldValue, Object newValue) {
+			if(connection.equals(ConnectionPropertySource.this.connection) && "extendedProperties".equals(property)) {
+				Display.getDefault().asyncExec(new Runnable() {
+					@Override
+					public void run() {
+						PropertySheet sh = null;
+						try {
+							sh = (PropertySheet)PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().findView("org.eclipse.ui.views.PropertySheet"); //$NON-NLS-1$
+						} catch (NullPointerException exc) {
```

Comment:
This lookup for the property sheet would be a great addition to OpenShiftUIUtils#showPropertySheet, centralizing the view id